frontend/packages/pipelines-plugin/src/components/pipelineruns/utils/tekton-results.ts

```diff
@@ -0,0 +1,345 @@
+              ? `data.metadata.labels["${expression.key}"] in [${expression.values.map(
+                  (value) => `"${value}"`,
+                )}]`
+              : '';
+          case 'Equals':
+            return expression.values?.[0]
+              ? EQ(`data.metadata.labels["${expression.key}"]`, expression.values?.[0])
+              : '';
+          case 'NotEquals':
+          case 'NotEqual':
+            return expression.values?.[0]
+              ? NEQ(`data.metadata.labels["${expression.key}"]`, expression.values?.[0])
+              : '';
+          case 'GreaterThan':
+            return expression.values?.[0]
+              ? EXP(`data.metadata.labels["${expression.key}"]`, expression.values?.[0], '>')
+              : '';
+          case 'LessThan':
+            return expression.values?.[0]
+              ? EXP(`data.metadata.labels["${expression.key}"]`, expression.values?.[0], '<')
+              : '';
+          default:
+            throw new Error(
+              `Tekton results operator '${expression.operator}' conversion not implemented.`,
+            );
+        }
+      })
+      .filter((x) => x),
+  );
+
+export const selectorToFilter = (selector) => {
+  let filter = '';
+  if (selector) {
+    const { matchLabels, matchExpressions, filterByName } = selector;
+
+    if (filterByName) {
+      filter = AND(filter, nameFilter(filterByName as string));
+    }
+
+    if (matchLabels || matchExpressions) {
+      if (matchLabels) {
+        filter = AND(filter, labelsToFilter(matchLabels));
+      }
+      if (matchExpressions) {
+        filter = AND(filter, expressionsToFilter(matchExpressions));
+      }
+    } else {
+      filter = labelsToFilter(selector as MatchLabels);
+    }
+  }
+  return filter;
+};
+
+// Devs should be careful to not cache a response that may not be complete.
+// In most situtations, caching is unnecessary.
+// Only cache a response that returns a single complete record as lists can change over time.
+let CACHE: { [key: string]: [any[], RecordsList] } = {};
+export const clearCache = () => {
+  CACHE = {};
+};
+const InFlightStore: { [key: string]: boolean } = {};
+
+let tektonResultUrl: string;
+
+export const createTektonResultsUrl = async (
+  namespace: string,
+  dataType: DataType,
+  filter?: string,
+  options?: TektonResultsOptions,
+  nextPageToken?: string,
+): Promise<string> => {
+  if (!tektonResultUrl) {
+    const tektonResult: K8sResourceKind = await k8sGetResource({
+      model: TektonResultModel,
+      name: 'result',
+    });
+    const targetNamespace = tektonResult?.spec?.targetNamespace;
+    const route: K8sResourceKind = await k8sGetResource({
+      model: RouteModel,
+      name: 'tekton-results-api-service',
+      ns: targetNamespace,
+    });
+    tektonResultUrl = route?.spec.host;
+  }
+  const URL = `https://${tektonResultUrl}/apis/results.tekton.dev/v1alpha2/parents/${namespace}/results/-/records?${new URLSearchParams(
```

Comment:
If there is no route.spec.host this will create an URL with undefined. Let us throw an error here or in the if statement above.

And use `const url` instead of `const URL` for a local variable.

```suggestion
  if (!tektonResultUrl) {
    throw new Error("...")
  }
  const url = `https://${tektonResultUrl}/apis/results.tekton.dev/v1alpha2/parents/${namespace}/results/-/records?${new URLSearchParams(
```